### PR TITLE
Screen recording implemented

### PR DESCRIPTION
**Pull request by YuryKebets**
## Change list

Screen recording implemented
 
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

### CODE REVIEW

File: appium-dotnet-driver/Appium/Android/AndroidStartScreenRecordingOptions.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        /// <summary>
+        /// Makes the recorder to display an additional information on the video overlay,
+        /// such as a timestamp, that is helpful in videos captured to illustrate bugs.
+        /// This option is only supported since API level 27 (Android P).
+        /// </summary>
+        /// <param name="videoSize">self instance for chaining.</param>
+        /// <returns>self instance for chaining.</returns>
+        public AndroidStartScreenRecordingOptions EnableBugReport(string videoSize)
```
Comment (mykola-mokhnach) [inline appium-dotnet-driver/Appium/Android/AndroidStartScreenRecordingOptions.cs]: argument copypaste?

File: appium-dotnet-driver/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Parameters["forceRestart"] = true;
+            return (T) this;
+        }
+
+        /// <summary>
+        /// The remotePath upload option is the path to the remote location,
+        /// where the resulting video should be uploaded.
+        /// The following protocols are supported: http/https(multipart), ftp.
+        ///
+        /// Missing value(the default setting) means the content of the resulting
```
Comment (mykola-mokhnach) [inline appium-dotnet-driver/Appium/ScreenRecording/BaseStartScreenRecordingOptions.cs]: missing space `value(the`

**Comment by mykola-mokhnach** [comment]
I would be happy to see some tests

File: integration_tests/iOS/IOSScreenRecordingTest.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            Thread.Sleep(1000);
+            string result = driver.StopRecordingScreen();
+            Assert.IsNotEmpty(result);
+        }
+
+        [Test]
+        public void ScreenRecordWithOptionsTest()
+        {
+            driver.StartRecordingScreen(
+                StartScreenRecordingOptions()
```
Comment (mykola-mokhnach) [inline integration_tests/iOS/IOSScreenRecordingTest.cs]: shouldn't these be iosrecordingoptions?

File: integration_tests/helpers/Caps.cs [hunk 4]
```diff
@@ -57,6 +57,16 @@ public static AppiumOptions getAndroid19Caps(string app)
             return capabilities;
         }
 
+        public static AppiumOptions getAndroid27Caps(string app)
+        {
+            AppiumOptions capabilities = new AppiumOptions();
+            capabilities.AddAdditionalCapability(CapabilityType.BrowserName, "");
```
Comment (mykola-mokhnach) [inline integration_tests/helpers/Caps.cs]: browser name should not be set if app is present

File: integration_tests/Android/AndroidScreenRecordingTest.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            Thread.Sleep(1000);
+            string result = driver.StopRecordingScreen();
+            Assert.IsNotEmpty(result);
+        }
+        
+        [Test]
+        public void ScreenRecordWithOptionsTest()
+        {
+            driver.StartRecordingScreen(
+                StartScreenRecordingOptions()
```
Comment (mykola-mokhnach) [inline integration_tests/Android/AndroidScreenRecordingTest.cs]: AndroidOptions?

File: appium-dotnet-driver/Appium/ScreenRecording/BaseScreenRecordingOptions.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        /// <summary>
+        /// Upload options set for the recorded screen capture.
+        /// </summary>
+        /// <param name="uploadOptions">Upload options</param>
+        /// <returns></returns>
+        public T WithUploadOptions(ScreenRecordingUploadOptions uploadOptions)
+        {
+            foreach (var parameter in uploadOptions.GetParameters())
```
Comment (mykola-mokhnach) [inline appium-dotnet-driver/Appium/ScreenRecording/BaseScreenRecordingOptions.cs]: I'm not an expert in .net, but I believe there should be a shortcut for such basic operation like dictionary cloning

```
var newDictionary = oldDictionary.ToDictionary(entry => entry.Key, entry => entry.Value);
```

Comment (YuryKebets) [reply appium-dotnet-driver/Appium/ScreenRecording/BaseScreenRecordingOptions.cs]: In this example, a new dictionary is created, but I need to add some elements to the existing dictionary.

**Comment by akinsolb** [comment]
closes #251 #269

**Pull request merged**